pkg/collector/parser/receiver.go

```diff
@@ -77,8 +77,49 @@ func IsRegistered(name string) bool {
 var (
 	endpointKey      = "endpoint"
 	listenAddressKey = "listen_address"
+	scraperReceivers = []string{
```

Comment:
Let's make `scraperReceivers` a `map[string]bool` so the lookup is is more efficient.